components/raftstore/src/store/peer.rs

```diff
@@ -2322,6 +2336,16 @@ where
                             "region_id" => self.region_id,
                         );
                     }
+                    // After the leadership changed, send `CasualMessage::Campaign` to the target
+                    // peer to campaign leader if there exists uncampaigned regions. It's used to
+                    // ensure that a leader is elected promptly for the newly created Raft
+                    // group, minimizing availability impact (e.g. #12410 and #17602.).
+                    for new_region in self.uncampaigned_new_regions.0.drain(..) {
+                        let _ = ctx.router.send(
+                            new_region,
+                            PeerMsg::CasualMessage(Box::new(CasualMessage::Campaign)),
```

Comment:
After sending the Compaign messages, we should clear the uncampaigned list?

Reply:
Yep, as it uses local msgs to notify the newly created region to `campaign`, it's safe to clear its list.